### PR TITLE
Update: Parse filters support multiple filters

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
The parser would incorrectly parse `dim|id-in[value,value 2],dim2|id-in[this]` as
```js
[
  { dimension: 'dim', field: 'id', operator: 'in', values: ['value', 'value 2'] },
  { dimension: ',dim2', field: 'id', operator: 'in', values: ['this'] }
]
```
**,dim2** and not **dim2**

## Proposed Changes
- Fix how we split the input
- added tests

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/mirage/routes/bard-lite-parsers.js [hunk 1]
```diff
@@ -34,8 +34,11 @@ export function parseSingleFilter(serializedFilter) {
  * @returns {Array} filters
  */
 export function parseFilters(filtersParam = '') {
-  const filterStrings = filtersParam
-    .split(']')
+  if (!filtersParam || filtersParam === '') {
```
Comment (jkusa) [inline packages/data/addon/mirage/routes/bard-lite-parsers.js]:
```suggestion
  if (!filtersParam) {
```

File: packages/data/addon/mirage/routes/bard-lite-parsers.js [hunk 2]
```diff
@@ -34,8 +34,11 @@ export function parseSingleFilter(serializedFilter) {
  * @returns {Array} filters
  */
 export function parseFilters(filtersParam = '') {
```
Comment (jkusa) [inline packages/data/addon/mirage/routes/bard-lite-parsers.js]: Do we want the default?

Comment (kevinhinterlong) [reply packages/data/addon/mirage/routes/bard-lite-parsers.js]:
```suggestion
export function parseFilters(filtersParam) {
```
No longer needed, 👍

File: packages/data/addon/mirage/routes/bard-lite-parsers.js [hunk 3]
```diff
@@ -33,9 +33,12 @@ export function parseSingleFilter(serializedFilter) {
...
  * @returns {Array} filters
  */
-export function parseFilters(filtersParam = '') {
-  const filterStrings = filtersParam
-    .split(']')
+export function parseFilters(filtersParam) {
+  if (!filtersParam) {
+    return [];
+  }
+  const filterStrings = (filtersParam + ',')
```
Comment (balajimanoharan) [inline packages/data/addon/mirage/routes/bard-lite-parsers.js]: template string 
```suggestion
  const filterStrings = `${filters},`
```

**Pull request merged**